apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -529,6 +545,16 @@ type RolloutStrategy struct {
 	// +kubebuilder:validation:Enum=RollingUpdate;External
 	Type RolloutStrategyType `json:"type,omitempty"`
 
+	// ResourceSnapshotIndex indicates the specific resource snapshot index to be rolled out to the target clusters.
+	// If not specified, the latest resource snapshot will be used.
+	// When the specified index does not exist, the behavior depends on the ResourceSnapshotCreationPolicy:
+	// - For the "Automatic" policy, the rollout will fail with reason ResourceSnapshotNotFound.
+	// - For the "OnDemand" policy, a new resource snapshot will be created using the current index if
+	// it is greater than the latest resourceSnapshot index.
+	// This field is only valid when RolloutStrategyType = RollingUpdate.
+	// +kubebuilder:validation:Optional
+	ResourceSnapshotIndex *string `json:"resourceSnapshotIndex,omitempty"`
```

Comment:
what if the ResourceSnapshotCreationPolicyType is ondemand? How would a user trigger a snapshot?

Reply:
when cx wants to cut the release, they update this field to trigger the snapshot creation.